File: infra/src/main/java/mixter/infra/PersistingEventPublisher.java
package mixter.infra;

import mixter.domain.Event;
import mixter.domain.EventPublisher;

public class PersistingEventPublisher implements EventPublisher {
    private final EventStore store;
    private final EventPublisher publisher;

    public PersistingEventPublisher(EventStore store, EventPublisher publisher) {
        this.store = store;
        this.publisher = publisher;
    }

    @Override
    public void publish(Event event) {
        store.store(event);
        publisher.publish(event);
    }
}

File: infra/src/main/scala/mixter/infra/PersistingEventPublisher.scala
package mixter.infra

import mixter.domain.{Event, EventPublisher}

class PersistingEventPublisher(eventStore: EventStore, eventPublisher: EventPublisher) extends EventPublisher {

  override def publish[T <: Event](event: T): Unit = {
    eventStore.store(event)
  }
}
